projects/driver_controls/src/magnetic_sensor.c

```diff
@@ -0,0 +1,63 @@
+#include "magnetic_sensor.h"
+#include "tlv493d.h"
+#include "delay.h"
+
+// Registers need to be read out and stored so that configuration registers aren't
+// accidentally overwritten on write
+static uint8_t s_read_reg[NUM_TLV493D_READ_REGISTERS];
+static uint8_t s_write_reg[NUM_TLV493D_WRITE_REGISTERS];
+
+// Returns reading in microteslas. Conversion method described in chapter 3.1 of the datasheet
+static int16_t prv_flux_conversion(uint8_t msb, uint8_t lsb) {
+  uint16_t data = (msb << 4) | (lsb & 0xF);
+
+  if (data > 1024) {
+    data -= 4096;
```

Comment:
Are you sure this is what you want to do? If you're trying to extend the sign, you're using an unsigned integer and it should probably be 2048. Also, if it is sign extension please comment it.